sdk/inc/azure/iot/az_iot_pnp_client.h

```diff
@@ -0,0 +1,670 @@
+ *
+ * @return An #az_result value indicating the result of the operation.
+ */
+AZ_NODISCARD AZ_INLINE az_result az_iot_pnp_client_get_sas_password(
+    az_iot_pnp_client const* client,
+    uint64_t token_expiration_epoch_time,
+    az_span base64_hmac_sha256_signature,
+    az_span key_name,
+    char* mqtt_password,
+    size_t mqtt_password_size,
+    size_t* out_mqtt_password_length)
+{
+  return az_iot_hub_client_sas_get_password(
+      &client->_internal.iot_hub_client,
+      token_expiration_epoch_time,
+      base64_hmac_sha256_signature,
+      key_name,
+      mqtt_password,
+      mqtt_password_size,
+      out_mqtt_password_length);
+}
+
+/*
+ *
+ * PnP Telemetry APIs
+ *
+ */
+
+/**
+ * @brief Gets the MQTT topic that must be used for device to cloud telemetry messages.
+ * @remark Telemetry MQTT Publish messages must have QoS At Least Once (1).
+ * @remark This topic can also be used to set the MQTT Will message in the Connect message.
+ *
+ * @param[in] client The #az_iot_pnp_client to use for this call.
+ * @param[in] component_name An #az_span specifying the component name to publish telemetry on.
+ * @param[in] properties Properties to attach to append to the topic.
+ * @param[out] mqtt_topic A buffer with sufficient capacity to hold the MQTT topic. If
+ *                        successful, contains a null-terminated string with the topic that
+ *                        needs to be passed to the MQTT client.
+ * @param[in] mqtt_topic_size The size, in bytes of \p mqtt_topic.
+ * @param[out] out_mqtt_topic_length __[nullable]__ Contains the string length, in bytes, of
+ *                                                  \p mqtt_topic. Can be `NULL`.
+ *
+ * @pre \p client must not be `NULL`.
+ * @pre \p mqtt_topic must not be `NULL`.
+ * @pre \p mqtt_topic_size must be greater than 0.
+ *
+ * @return An #az_result value indicating the result of the operation.
+ * @retval #AZ_OK The topic was retrieved successfully.
+ */
+AZ_NODISCARD az_result az_iot_pnp_client_telemetry_get_publish_topic(
+    az_iot_pnp_client const* client,
+    az_span component_name,
+    az_iot_message_properties* properties,
+    char* mqtt_topic,
+    size_t mqtt_topic_size,
+    size_t* out_mqtt_topic_length);
+
+/*
+ *
+ * PnP Command APIs
+ *
+ */
+
+/**
+ * @brief The MQTT topic filter to subscribe to command requests.
+ * @remark Commands MQTT Publish messages will have QoS At most once (0).
+ */
+#define AZ_IOT_PNP_CLIENT_COMMANDS_SUBSCRIBE_TOPIC "$iothub/methods/POST/#"
+
+/**
+ * @brief A command request received from IoT Hub.
+ *
+ */
+typedef struct
+{
+  az_span
+      request_id; /**< The request id.
+                   * @note The application must match the command request and command response. */
+  az_span component; /**< The name of the component which the command was invoked for.
+                      * @note Can be `AZ_SPAN_EMPTY` if for the root component */
+  az_span name; /**< The command name. */
```

Comment:
>name [](start = 10, length = 4)

command name?

Reply:
should these be `component_name` and `command_name`?

Reply:
Those sound good to me.